### PR TITLE
Added Gradle Android Library Example

### PR DESCRIPTION
**Pull request by sebasira**
This is an example to show that `.aar` artifacts are not uploaded to CI Server. As requested in https://github.com/jfrog/jfrog-cli/issues/880

### CODE REVIEW

**Comment by sebasira** [comment]
I have read the CLA Document and I hereby sign the CLA

File: gradle-examples/gradle-android-library-example-ci-server/app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,47 @@
+apply plugin: 'com.android.library'
```
Comment (yahavi) [inline gradle-examples/gradle-android-library-example-ci-server/app/build.gradle:1]:
```suggestion
apply plugin: 'com.android.library'
apply plugin: 'maven-publish'
```

File: gradle-examples/gradle-android-library-example-ci-server/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    libraryVariants.all { variant ->
+        variant.outputs.all {
+            // Do not add Variant (buildType) Name at the end
+            outputFileName = "${rootProject.name}-${defaultConfig.versionName}.aar"
+        }
+    }
+}
+
+dependencies {
+
```
Comment (yahavi) [inline gradle-examples/gradle-android-library-example-ci-server/app/build.gradle:41]:
```suggestion
```

File: gradle-examples/gradle-android-library-example-ci-server/app/build.gradle [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    // Note that 'defaultConfig' references the above closure with that name
+    // And rootProject.name is defined in settings.gradle
+    libraryVariants.all { variant ->
+        variant.outputs.all {
+            // Do not add Variant (buildType) Name at the end
+            outputFileName = "${rootProject.name}-${defaultConfig.versionName}.aar"
+        }
+    }
+}
+
```
Comment (yahavi) [inline gradle-examples/gradle-android-library-example-ci-server/app/build.gradle:39]:
```suggestion

afterEvaluate {
    publishing {
        publications {
            release(MavenPublication) {
                from components.release
                artifactId = "${rootProject.name}"
                version = "${android.defaultConfig.versionName}"
            }
        }
    }
    artifactoryPublish {
        publications(publishing.publications.release)
    }
}

```

File: gradle-examples/gradle-android-library-example-ci-server/app/build.gradle [hunk 4]
```diff
@@ -0,0 +1,47 @@
+apply plugin: 'com.android.library'
+
+android {
+    compileSdkVersion 30
+    buildToolsVersion "30.0.3"
+
+    defaultConfig {
+        applicationId "com.example.testlibrary"
```
Comment (yahavi) [inline gradle-examples/gradle-android-library-example-ci-server/app/build.gradle:8]:
```suggestion
```

I'm not sure that a library can have an application ID: 
> com.android.builder.errors.EvalIssueException: Library projects cannot set applicationId. applicationId is set to 'com.example.testlibrary' in default config.

**Review by yahavi — CHANGES_REQUESTED** [review]
@sebasira 
Thanks for this example. I added in comments the required changes to make the example publish to Artifactory. 
In order to help others, would you like to push the changes to merge this example?

File: gradle-examples/gradle-android-library-example-ci-server/app/build.gradle (hunk 4 above)
Comment (sebasira) [reply gradle-examples/gradle-android-library-example-ci-server/app/build.gradle:9]: You're correct! Only applications (not libraries) can set `applicationId`. My bad!

**Comment by sebasira** [comment]
I have approve all of your changes. Do you need more from me to merge this request?

**Comment by yahavi** [comment]
@sebasira,
Actually, I think that adding the gitlab file is not necessary, since we want this example to be as generic as possible.

File: gradle-examples/gradle-android-library-example-ci-server/app/build.gradle [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            release(MavenPublication) {
+                from components.release
+                artifactId = "${rootProject.name}"
+                version = "${android.defaultConfig.versionName}"
+            }
+        }
+    }
+    artifactoryPublish {
+        publications(publishing.publications.release)
+    }
```
Comment (yahavi) [inline gradle-examples/gradle-android-library-example-ci-server/app/build.gradle:52]: Let's wrap it with `pluginManager.withPlugin('com.jfrog.artifactory')` to make it work when running without the JFrog CLI, as discussed in https://github.com/jfrog/jfrog-cli/issues/880#issuecomment-898989046
```suggestion
    pluginManager.withPlugin('com.jfrog.artifactory') {
        artifactoryPublish {
            publications(publishing.publications.release)
        }
    }
```

Comment (sebasira) [reply gradle-examples/gradle-android-library-example-ci-server/app/build.gradle:52]: Yes! This is what I like the most!

**Comment by sebasira** [comment]
> @sebasira,
> Actually, I think that adding the gitlab file is not necessary, since we want this example to be as generic as possible.

OK! I'm ok with that, I will remove it

**Comment by yahavi** [comment]
Thanks, @sebasira!

**Pull request merged**